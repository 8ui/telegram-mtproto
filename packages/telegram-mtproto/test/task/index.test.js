
// import raw from './example-raw'
// import { parserOpts } from './example-binary'
// jest.mock('../../lib/config-provider')
// jest.mock('../../src/config-provider')
// require('../../src/config-provider')

import { api } from '../mtproto-config'

beforeAll(() => {
  jest.mock('../../lib/config-provider/provider')
  const { MTProto } = require('../../lib')
  MTProto({ server: {
    dev     : true,
    webogram: true
  }, api })
})

// import parser from '../../lib/service/chain'

// import fullConfig, { rawData } from './example-normalize'


afterAll(() => {
  jest.unmock('../../lib/config-provider/provider')
})



// import normalize from '../../lib/task/new-index'

// import target from './example-target'





// test('full test', async() => {
//   // console.log(client.cache)
//   await expect(normalize(rawData)).resolves.toEqual(target)
// })

test('binary parsing', async() => {
  // await expect(parser(parserOpts)).resolves.toMatchSnapshot()
  const parser = require('../../lib/service/chain').default
  const fullConfig = require('./example-normalize').default
  console.log('a')
  await expect(parser(fullConfig)).resolves.toMatchSnapshot()
})

