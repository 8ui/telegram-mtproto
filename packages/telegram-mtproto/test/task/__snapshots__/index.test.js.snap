// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`binary parsing 1`] = `
Object {
  "messageID": "6450609992032546817",
  "response": Object {
    "_": "msg_container",
    "messages": Array [
      Object {
        "_": "message",
        "body": Object {
          "_": "new_session_created",
          "first_msg_id": "6450609986432292820",
          "server_salt": "16751524301008951348",
          "unique_id": "1563096998223171911",
        },
        "bytes": 28,
        "msg_id": "6450609989367164929",
        "seqno": 1,
      },
      Object {
        "_": "message",
        "body": Object {
          "_": "rpc_result",
          "req_msg_id": "6450609986432292820",
          "result": Object {
            "_": "auth.sentCode",
            "flags": 3,
            "next_type": Object {
              "_": "auth.codeTypeSms",
            },
            "phone_code_hash": "1f7f88db961d8ee62f",
            "phone_registered": true,
            "type": Object {
              "_": "auth.sentCodeTypeApp",
              "length": 5,
            },
          },
        },
        "bytes": 52,
        "msg_id": "6450609989668679681",
        "seqno": 3,
      },
    ],
  },
  "seqNo": 4,
  "sessionID": Uint8Array [
    89,
    35,
    220,
    168,
    231,
    136,
    44,
    110,
  ],
}
`;
